File: 2244. Minimum Rounds to Complete All Tasks/Solution2244.cpp
#include<iostream>
#include<vector>
#include<unordered_map>

using namespace std;

class Solution2244 {
public:
    int minimumRounds(vector<int>& tasks) {
        unordered_map<int, int> counter;
        for (int t : tasks) {
            counter[t]++;
        }
        int res = 0;
        for (auto& it : counter) {
            if (it.second == 1) return -1;
            res += (it.second + 2) / 3;
        }
        return res;
    }
};

File: 2244. Minimum Rounds to Complete All Tasks/Solution2244.py
from typing import Counter, List


class Solution2244:
    def minimumRounds(self, tasks: List[int]) -> int:
        freq = Counter(tasks).values()
        return -1 if 1 in freq else sum((i+2)//3 for i in freq)

tasks = [2,2,3,3,2,4,4,4,4,4]
sol = Solution2244()
res = sol.minimumRounds(tasks)
print(res)